modules/game/src/test/EventTest.scala

```diff
@@ -31,3 +32,98 @@ class EventTest extends munit.ScalaCheckSuite:
     Event.PossibleMoves.json(moves) match
       case JsString(str) => str
       case _             => failSuite("Expected JsString")
+
+  test("Enpassant anti regression"):
+    var event = Event.Enpassant(pos = chess.Square.A1, color = chess.Color.White)
+    assertEquals(event.typ, "enpassant")
+    assertEquals(
+      event.data,
+      Json.obj(
+        "key"   -> "a1",
+        "color" -> "white"
+      )
+    )
+    event = Event.Enpassant(pos = chess.Square.C3, color = chess.Color.Black)
+    assertEquals(event.typ, "enpassant")
+    assertEquals(
+      event.data,
+      Json.obj(
+        "key"   -> "c3",
+        "color" -> "black"
+      )
+    )
```

Comment:
I don't think we should be testing such trivial code. Look:

```scala
  case class Enpassant(pos: Square, color: Color) extends Event:
    def typ = "enpassant"
    def data =
      Json.obj(
        "key"   -> pos.key,
        "color" -> color
      )
```
What room is there for bugs, and does it justify 15 lines of tests? I don't think so personally.

Reply:
Thank you for the direction Le.

I've added some property based tests, like you mentioned, a test for `Event.Move` which is one of the most complicated cases.

I assume you want to keep the "anti regression" type tests, and also have the prop-based ones?

Reply:
So it looks like with prop-based testing I'm replicating the code in the tests (e.g. in `Move` `check` is added only if it's true).

Happy to go with whatever approach you want to, but it seems like the "anti regression" style tests are simpler and more robust.